### ISSUE TITLE
Configuring rule exclusions

### ISSUE THREAD

**Issue by ichirugame** [coreruleset/coreruleset#3060]
I don't know how to set exceptions for rules.
I am using nginx.

**Comment by airween** [comment]
Hi @ichirugame,

thanks for contacting us.

Please review our [issue template](https://github.com/coreruleset/coreruleset/issues/new?assignees=&labels=False+Positive&template=01_false-positive.md&title=) for cases like this, and complete all fields.

Without that, I'm afraid we can't help you.

**Comment by airween** [comment]
If you are using CRS 3.3.4, then you can use its own exclusion rules for [Wordpress](https://github.com/coreruleset/coreruleset/blob/v3.3/master/rules/REQUEST-903.9002-WORDPRESS-EXCLUSION-RULES.conf). As you can see in lines [20](https://github.com/coreruleset/coreruleset/blob/v3.3/master/rules/REQUEST-903.9002-WORDPRESS-EXCLUSION-RULES.conf#L20) and [29](https://github.com/coreruleset/coreruleset/blob/v3.3/master/rules/REQUEST-903.9002-WORDPRESS-EXCLUSION-RULES.conf#L29), to activate this exclusion set you have to set up a transaction variable (at least). So if you want to activate this exclusion set only for one site, you should put a rule into your crs-setup.conf like this:

```
SecRule &REQUEST_HEADERS:Host "!@eq 0" \
    "id:990001,\
    phase:1,\
    pass,\
    t:none,\
    nolog,\
    chain"
    SecRule REQUEST_HEADERS:Host "@strEq your.host.name" \
        "setvar:tx.crs_exclusions_wordpress=1"
```

If you want to write a custom exclusion, you can write something like this:

```
SecRule &REQUEST_HEADERS:Host "!@eq 0" \
    "id:990001,\
    phase:1,\
    pass,\
    t:none,\
    nolog,\
    chain"
    SecRule REQUEST_HEADERS:Host "@strEq your.host.name" \
        "chain"
        "SecRule REQUEST_FILENAME "@endsWith /wp-login.php" \
            "ctl:ruleRemoveById=960024,\
            ctl:ruleRemoveById=980130,\
            ctl:ruleRemoveById=949110,\
            ctl:ruleRemoveById=941310"
```

Please try one of these, and let us know your result.

**Comment by ichirugame** [comment]
thank you.
To set multiple of URL
"SecRule REQUEST_FILENAME "@endsWith /wp-login.php | /wp-admin" \
Will the code look like this?

**Comment by airween** [comment]
> To set multiple of URL `SecRule REQUEST_FILENAME " /wp-login.php | /wp-admin"` Will the code look like this?

Well, if you do not set any operator, that means it's the `@rx`, and the argument can be interpreted as a regex, but I think this is much better:

```
SecRule REQUEST_FILENAME "@rx ^/(wp-login\.php|wp-admin)$"
```
Use the anchors wherever you can in case of `@rx` or use [@within](https://github.com/SpiderLabs/ModSecurity/wiki/Reference-Manual-(v2.x)#within), which checks the exact match.

And do not forget that if you don't check the `Hostname` this exclusion will work in case of all sites.
